templates/puppet_signscript.rb.erb

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env ruby
+#
+
+require 'net/http'
+require 'net/https'
+require 'fileutils'
+require 'json'
+require 'yaml'
+
+def log(message)
+  puts message
+  `logger "[PuppetCA CSR Verify] #{message}"`
+end
+
+csr = STDIN.read
+
+settings_file    = "<%= scope.lookupvar("foreman_proxy::etc") %>/foreman-proxy/settings.yml"
+SETTINGS         = YAML.load_file(settings_file)
+protocol         = "<% if scope.lookupvar("foreman_proxy::puppetca_listen_on") != "http" %>https<% else %>http<% end %>"
```

Comment:
I believe this should go into a settings file for the script. I don't like having a script as a template or users needing to make changes to this if they don't use the installer.

Reply:
Otherwise you can parse it from the foreman-proxy settings. The settings.d directory is a setting in settings.yaml and there you can load `puppetca`.